app/bundles/CoreBundle/Assets/js/1.core.js

```diff
@@ -135,9 +137,120 @@ MauticVars.intervalsInProgress   = {};
 
 var Mautic = {
     loadedContent: {},
-
     keyboardShortcutHtml: {},
 
+    /**
+     * Initializes dismissed elements by injecting necessary CSS.
+     */
+    initializeDismissedElements: function() {
+        // Ensure MauticVars and dismissedElements exist
+        this.dismissedElements = JSON.parse(localStorage.getItem('dismissedElements')) || [];
+        this.dismissedStyle = null;
+
+        if (this.dismissedElements.length > 0) {
+            // Combine IDs with commas for efficient CSS
+            var selector = this.dismissedElements.map(function(id) {
+                return '#' + id;
+            }).join(', ');
+
+            var css = selector + ' { display: none !important; }';
+
+            // Create a style element and append the CSS
+            this.dismissedStyle = document.createElement('style');
+            this.dismissedStyle.type = 'text/css';
+            this.dismissedStyle.appendChild(document.createTextNode(css));
+
+            // Append the style element to the document head
+            var head = document.head || document.getElementsByTagName('head')[0];
+            head.appendChild(this.dismissedStyle);
+        }
+    },
+
+    /**
+     * Dismisses an element by ID.
+     *
+     * @param {string} elementId - The ID of the element to dismiss.
+     */
+    dismissElement: function(elementId) {
+        if (this.dismissedElements.indexOf(elementId) === -1) {
+            this.dismissedElements.push(elementId);
+            localStorage.setItem('dismissedElements', JSON.stringify(this.dismissedElements));
+
+            // Inject CSS to hide the newly dismissed element
+            if (this.dismissedStyle) {
+                var newSelector = '#' + elementId;
+                this.dismissedStyle.appendChild(document.createTextNode(newSelector + ' { display: none !important; }'));
+            } else {
+                // Create a new style element if not existing
+                var css = '#' + elementId + ' { display: none !important; }';
+                this.dismissedStyle = document.createElement('style');
+                this.dismissedStyle.type = 'text/css';
+                this.dismissedStyle.appendChild(document.createTextNode(css));
+
+                // Append the style element to the document head
+                var head = document.head || document.getElementsByTagName('head')[0];
+                head.appendChild(this.dismissedStyle);
+            }
+
+            // Hide the element
+            var element = mQuery('#' + elementId);
+            if (element.length) {
+                element.hide();
+            }
+        }
+    },
+
+    /**
+     * Resets all dismissed elements.
+     */
+    resetDismissedElements: function() {
+        // Clear the dismissedElements array
+        this.dismissedElements = [];
+        localStorage.setItem('dismissedElements', JSON.stringify(this.dismissedElements));
```

Comment:
We shouldn't use local storage to store user preferences. Same for the theme and colors. There is already a database field in the `users` table called 'preferences` where we should store this information so it's not getting removed when user changes computers/browsers or the local storage gets cleared.

Can we hop on a call if you need a hand with how to do it.

Reply:
Why shouldn't we?

This way all logic is handled by the user side, locally, instead of depending on the server, as everything here is about UI

Reply:
I mentioned why:

> so it's not getting removed when user changes computers/browsers or the local storage gets cleared.

Mautic is a web app. So when you save something you expect that the state will be the same on the second device that you open the web app in. 

Another reason was shared by some other tester. It's inconsistent with any other Mautic form. The change is "stored" when you click, not when you click the "Save" button.

This is a big issue in UX. I'm kinda surprised that you don't see it :)

Reply:
@escopecz I'm mainly thinking about this: why should the user save something before the system starts remembering a dismissed tip?

Let's consider these points:
- Following your approach, we would create a dependency on whatever the user is doing on the screen (like having to save a contact so they don’t see the tip again)
- it might be less intuitive than just clicking on the X…
- it could become problematic on screens where there are no buttons to save changes, like a tip on an index page. What could be the trigger to save here?
- to abandon the localStorage approach, we'll also have to remove the clone campaign events functionality, as something you clone in the first device won't show on another (check campaign.js)
- we'd have to clear versioning-related code too, as they also use localStorage (check 13.versioning.js)

Another way could be to remove the dismissible functionality and make the tips shown every time, forever, but that would revert what you asked to improve previously

What do you think? Which way would you find better?

Reply:
INMO the UI of the TIP is OK. Probably @escopecz  mean something else

Reply:
@andersonjeccel I was talking about the backend templates and colors in the user profile where is the save button but it doesn't work for these 2 new features.

Since I see it again in this PR I think we should really think about storing the user-related preferences properly in the database.

Cloning a campaign event isn't something you'd do on different computers. But if you change a theme, you want it to stick. For example, yesterday I had a dark theme. I woke up today and it's back to default.

You are correct that the save button doesn't apply here. We can create an ajax call that will set the property to the user database record when dismissing the notification. What apply is that the user doesn't want to see the notification again when changing browser/device or the local storage gets cleared. I don't have such a problem with storing this information to the local storage. It's not a big deal. The user profile changes are. I'm just more vokal here because it's 3rd PR where I see you using local storage and there is a better way how to do it. The goal is that we join forces and prepare Mautic so you will be able to store values in the database.

Reply:
Using database instead of local storage is undisputable topic. We should store personal settings to database .
Switch themes and colors without save button is OK to me.

Reply:
@escopecz @kuzmany I'd need someone teching me how to do it until reach a desired solution

Or I can leave this way for someone to refactor later

What do you find better?